src/signals/incident/components/form/Header/index.test.js

```diff
@@ -1,93 +1,122 @@
 
-    expect(wrapper).toMatchSnapshot();
+    expect(queryByText('Het moet een geldig e-mailadres zijn')).toBeInTheDocument();
   });
 
-  describe('error messages', () => {
-    it('should render no error when element is only touched', () => {
-      wrapper.setProps({ touched: true });
+  it('should render maxLength error', () => {
+    const requiredLength = 300;
+    const hasError = prop => prop === 'maxLength';
+    const getError = () => ({ requiredLength });
 
-      expect(wrapper).toMatchSnapshot();
-    });
+    const { queryByText, rerender } = render(withAppContext(<Header hasError={() => false} getError={getError} touched />));
 
-    it('should render required error', () => {
-      hasError.mockImplementation(type => type === 'required');
-      wrapper.setProps({ touched: true });
+    expect(queryByText(`U kunt maximaal ${requiredLength} tekens invoeren.`)).not.toBeInTheDocument();
 
-      expect(wrapper).toMatchSnapshot();
-    });
+    rerender(withAppContext(<Header hasError={hasError} getError={getError} touched />));
 
-    it('should render email error', () => {
-      hasError.mockImplementation(type => type === 'email');
-      wrapper.setProps({ touched: true });
+    expect(queryByText(`U kunt maximaal ${requiredLength} tekens invoeren.`)).toBeInTheDocument();
+  });
+
+  it('should render custom error', () => {
+    const error = 'Hic sunt dracones';
+    const hasError = prop => prop === 'custom';
+    const getError = () => error;
 
-      expect(wrapper).toMatchSnapshot();
-    });
+    const { queryByText, rerender } = render(withAppContext(<Header hasError={() => false} getError={getError} touched />));
 
-    it('should render maxLength error', () => {
-      hasError.mockImplementation(type => type === 'maxLength');
-      getError.mockImplementation(() => ({
-        requiredLength: 666,
-      }));
-      wrapper.setProps({ touched: true });
+    expect(queryByText(error)).not.toBeInTheDocument();
 
-      expect(wrapper).toMatchSnapshot();
-    });
+    rerender(withAppContext(<Header hasError={hasError} getError={getError} touched />));
 
-    it('should render custom error', () => {
-      hasError.mockImplementation(type => type === 'custom');
-      getError.mockImplementation(() => 'custom error message');
-      wrapper.setProps({ touched: true });
+    expect(queryByText(error)).toBeInTheDocument();
+  });
 
-      expect(wrapper).toMatchSnapshot();
-    });
+  it('should not render error when not touched', () => {
+    const hasError = prop => prop === 'required';
+    const touched = false;
 
-    it('should render optional field message', () => {
-      wrapper.setProps({ options: { validators: [] } });
+    const { queryByText } = render(withAppContext(<Header hasError={hasError} touched={touched} />));
 
-      expect(wrapper).toMatchSnapshot();
-    });
+    expect(queryByText('Dit is een verplicht veld')).not.toBeInTheDocument();
```

Comment:
Where do you test the positive version?

Reply:
In the 'should render required error' test case

Reply:
In this case as well, put the string in a `const`, to make sure the negative test tests what it should.